packages/cli/src/cmds/validator/keymanager/impl.ts

```diff
@@ -149,7 +149,7 @@ export class KeymanagerApi implements Api {
 
         decryptKeystores.queue(
           {keystoreStr, password},
-          (secretKeyBytes: Uint8Array) => {
+          async (secretKeyBytes: Uint8Array) => {
```

Comment:
`decryptKeystores.queue` does no handle promise rejections. Why not decrypt all keystores first, accumulate the result of this callbacks to the queue, then on the main function body run the code below that can throw / reject?

Reply:
> decryptKeystores.queue does no handle promise rejections.

That's no issue, due to how promise chaining works. In any case sync or async this will be handled by `onError` handler
https://github.com/ChainSafe/lodestar/blob/1aa656191c84e3b55281bb8f21c9f4a529282500/packages/cli/src/cmds/validator/keymanager/decryptKeystores/threadPool.ts#L48

I was more concerned about the success case, i.e. does `await decryptKeystores.completed();` properly wait for all promises to be resolved but this is the case as well. I tested both those cases manually, also existing test cases should cover that.

> Why not decrypt all keystores first, accumulate the result of this callbacks to the queue

Don't we lose a lot of concurrency by doing that? Decrypt keystores is CPU heavy while write keystores and add signers are pretty lightweight I/O operations, I think it is good that we run that in parallel.

Also if we assume an import of 100+ keys it potentially takes a few minutes, best to add signers to validator store asap.